### PR TITLE
server: migrate test-infra to testify for `server/conn_test.go`

### LINKED ISSUE #27343: migrate test-infra to testify for `server/conn_test.go`
This is a subtask of #26863

### PR DESCRIPTION
**Pull request by yedamao**
### What problem does this PR solve?

Issue Number: close #27343  

### Release note 

```release-note
None.
```

### CODE REVIEW

**Review by tisonkun — CHANGES_REQUESTED** [review]
The direction of this pr is in a wrong direction. Please take a look at https://github.com/pingcap/tidb/pull/27377 as a demo how to migrate suite.

After this patch, there is still "pingcap/check" dependency and leaktest...

**Comment by tisonkun** [comment]
This suite can fit the pattern `testkit.CreateMockStore`, just use it.

**Comment by yedamao** [comment]
> After this patch, there is still "pingcap/check" dependency and leaktest...

Some tests remains in conn_stmt_test.go, when the migration is complete, it will be removed.

**Comment by yedamao** [comment]
> This suite can fit the pattern `testkit.CreateMockStore`, just use it.
> 
> You can extend the original function with a `MockTiKVStoreOption` parameter.

I checked the `testkit.CreateMockStore`  and #27377. 
The `testkit.CreateMockStore` just return a mockstore, but some tests also depend on `domain.Domain`.

**Comment by tisonkun** [comment]
@yedamao be aware that you are a TiDB developer. You can always change the function on demand if reasonable.

File: server/conn_test.go [hunk 1]
```diff
@@ -47,6 +50,32 @@ type ConnTestSuite struct {
...
+		}),
+		mockstore.WithStoreType(mockstore.EmbedUnistore),
+	)
+	require.NoError(t, err)
+	dom, err := session.BootstrapSession(store)
+	require.NoError(t, err)
+
+	return &connTestHelper{store: store, dom: dom}
+}
+
```
Comment (yedamao) [inline server/conn_test.go:78]: @tisonkun How about move this connTestHelper to testkit pkg as a common test helper.

**Comment by yedamao** [comment]
@tisonkun Please take a look at 9487285

**Comment by tisonkun** [comment]
> @tisonkun Please take a look at [9487285](https://github.com/pingcap/tidb/commit/94872850f824a1f01fae92f110659529f58e4abf)

It looks in a right direction.

**Review by tiancaiamao — APPROVED** [review]

**Review by xhebox — APPROVED** [review]

**Comment by tisonkun** [comment]
/run-check_dev_2

emm...

[2021-08-31T03:56:07.849Z] The following test cases take too long to finish:
[2021-08-31T03:56:07.849Z] PASS: tidb_test.go:878: tidbTestSerialSuite.TestTLSAuto	5.054s

**Pull request merged**